src/mito2/src/memtable/merge_tree/data.rs

```diff
@@ -519,26 +522,137 @@ impl<'a> DataPartEncoder<'a> {
             true,
         )?;
         writer.write(&rb).context(error::EncodeMemtableSnafu)?;
-        let _file_meta = writer.close().context(error::EncodeMemtableSnafu)?;
-        Ok(Bytes::from(bytes))
+        let metadata = writer.close().context(error::EncodeMemtableSnafu)?;
+        Ok(DataPart::Parquet(ParquetPart {
+            data: Bytes::from(bytes),
+            metadata,
+        }))
     }
 }
 
+/// Data parts under a shard.
+pub struct DataParts {
```

Comment:
We need to add a data buffer to the `DataParts` as we will write rows into the `DataParts`.

Reply:
> We need to add a data buffer to the `DataParts` as we will write rows into the `DataParts`.

fixed in ac863a7a33